src/read/archive.rs

```diff
@@ -93,6 +89,40 @@ impl<'data, R: ReadRef<'data>> ArchiveFile<'data, R> {
                     file.symbols = (symtbl, len);
                 }
             }
+
+            // Big archive member table lists file entries with offsets and names. To avoid
+            // potential infinite loop (members are double-linked list), the iterator goes
+            // through the table instead of real members.
+            let mut member_table_offset = parse_u64_digits(&file_header.memoff, 10)
+                .read_error("Invalid offset for member table of AIX big archive")?;
+            if member_table_offset > 0 {
+                // The table itself is also a file with header.
+                let header = data
+                    .read::<archive::AixHeader>(&mut member_table_offset)
+                    .read_error("Invalid AIX big archive member header")?;
+                let terminator = data
+                    .read_bytes(&mut member_table_offset, 2)
+                    .read_error("Invalid AIX big archive terminator")?;
+                if terminator != archive::TERMINATOR {
+                    return Err(Error("Invalid AIX big archive terminator"));
+                }
+
+                let nummem = data
+                    .read_bytes_at(member_table_offset, 20)
+                    .read_error("Invalid member count in AIX big archive")?;
+                let members_count = parse_u64_digits(nummem, 10)
+                    .read_error("Invalid member count in AIX big archive")?;
+                let table_size = parse_u64_digits(&header.size, 10)
+                    .read_error("Invalid AIX big archive member size")?;
+                file.offset = member_table_offset + 20;
+                file.len = file.offset + members_count * 20;
+                file.names = data.read_bytes_at(file.len, table_size - (members_count + 1) * 20)
```

Comment:
Since `members_count` is from data in the file, it would be possible for it to be invalid and cause overflows in this arithmetic. We need to check for that and return errors.